### DISCUSSION TITLE
"Created ... ago" self updating Label

### DISCUSSION THREAD

**Discussion by mfriese** [AvaloniaUI/Avalonia#5806] [Q&A] [answered]
Hi there,
I'd like to create a Label that shows, when an item was created. Via binding I get a DateTime object, _which will always be the same_ for my item. But in my frontend I'd like to change the displayed text (I use a converter to convert the DateTime object to readable string) to something like this:
- Created seconds ago
- Created minutest ago
- Created hours ago
- Created days ago
depending on how much time passed since creation. I don't really know how to update the UI without there being any changes to the data. What would be the best way to updated the displayed text?

**Comment by kekekeks** [comment, marked as answer]
You probably need a custom user control with CreatedAt property. Define a static timer and subscribe/unsubscribe to it from OnAttachedToVirtualTree/OnDetachedFromVirtualTree, then update the label text if necessary.

**Reply by mfriese** [reply]
Thank you! So I've created a custom control with a new property and a timer. The timer would then update the content property every n seconds. Works fine, thank you!
